src/stan_math_backend/Expression_gen.ml

```diff
@@ -190,14 +190,23 @@ and read_data ut ppf es =
   pf ppf "context__.vals_%s(%a)" i_or_r pp_expr (List.hd_exn es)
 
 and gen_distribution_app f =
+  let ifx = Middle.proportional_to_distribution_infix in
   if
+    String.is_suffix f ~suffix:(ifx ^ "_lpmf")
+    || String.is_suffix f ~suffix:(ifx ^ "_lpdf")
+    || String.is_suffix f ~suffix:(ifx ^ "_log")
+  then
+    Some
+      (fun ppf ->
+        let f = String.substr_replace_first ~pattern:ifx ~with_:"" f in
```

Comment:
@seantalts , I just realized that this doesn't quite do the right thing in some edge cases. Could you maybe fix it? It could be that a function name has "_propto" in it earlier than in the suffix, technically. We don't want to replace the first, but we want to replace the last.

Reply:
Hmm, there is no `substr_replace_last`, and I think it would be very confusing to, at this point, create distributions with "_propto" in their name (possibly we should disallow this for UDFs?). I guess I can reverse it, replace first, reverse it again :P